src/Avalonia.Animation/AnimationInstance`1.cs

```diff
@@ -73,17 +71,19 @@ public AnimationInstance(Animation animation, Animatable control, Animator<T> an
             _fillMode = animation.FillMode;
             _onCompleteAction = OnComplete;
             _interpolator = Interpolator;
+            _baseClock = baseClock;
         }
 
         protected override void Unsubscribed()
         {
             _timerSubscription?.Dispose();
+            _clock.PlayState = PlayState.Stop;
         }
 
         protected override void Subscribed()
         {
-            _timerSubscription = Timing.AnimationsTimer
-                                       .Subscribe(p => this.Step(p));
+            _clock = new Clock(_baseClock);
```

Comment:
Why does a new clock need to be created here? What would happen if `_baseClock` were used?

Reply:
We would have to track the time of the first frame that the `AnimationInstance` sees and use that as our base time. I thought that since clock instances are relatively lightweight that it would be easier to use derived clock instances in `AnimationInstance` and `TransitionInstance` so the clock itself would do all of the timekeeping math.

Reply:
Ok, that makes sense. It's an implementation detail anyway so can be easily changed if it's causing a lot of clocks to be created.